pkg/storage/ingest/reader.go

```diff
@@ -437,22 +408,73 @@ func (r *PartitionReader) newKafkaReader(at kgo.Offset) (*kgo.Client, error) {
 	return client, nil
 }
 
-func (r *PartitionReader) fetchLastCommittedOffsetWithRetries(ctx context.Context) (offset int64, exists bool, err error) {
-	var (
-		retry = backoff.New(ctx, backoff.Config{
-			MinBackoff: 100 * time.Millisecond,
-			MaxBackoff: 2 * time.Second,
-			MaxRetries: 10,
-		})
-	)
+func (r *PartitionReader) getStartOffset(ctx context.Context) (startOffset, lastConsumedOffset int64, err error) {
+	switch r.kafkaCfg.ConsumeFromPositionAtStartup {
+	case consumeFromStart:
+		startOffset = kafkaOffsetStart
+		level.Info(r.logger).Log("msg", "starting consumption from partition start", "start_offset", startOffset, "consumer_group", r.consumerGroup)
+		return startOffset, -1, nil
+
+	case consumeFromEnd:
+		startOffset = kafkaOffsetEnd
+		level.Warn(r.logger).Log("msg", "starting consumption from partition end (may cause data loss)", "start_offset", startOffset, "consumer_group", r.consumerGroup)
+		return startOffset, -1, nil
+	}
+
+	// We use an ephemeral client to fetch the offset and then create a new client with this offset.
+	// The reason for this is that changing the offset of an existing client requires to have used this client for fetching at least once.
+	// We don't want to do noop fetches just to warm up the client, so we create a new client instead.
+	cl, err := kgo.NewClient(commonKafkaClientOptions(r.kafkaCfg, r.metrics.kprom, r.logger)...)
+	if err != nil {
+		return 0, -1, fmt.Errorf("unable to create bootstrap client: %w", err)
+	}
+	defer cl.Close()
 
+	// lastConsumedOffset is unknown until (and only if) we successfully fetch its value below.
+	lastConsumedOffset = -1
```

Comment:
[non blocking] From a design perspective, to me it looks a bit inconsistent that `fetchOffset()` updates `lastConsumedOffset` (whose scope is outside the function) but returns the start offset. I think `fetchOffset()` could return `lastConsumedOffset` too.